test/run

```diff
@@ -6,6 +6,8 @@ if [ -z "${SKIP_STATIC_CHECK:-}" ]; then
 	# run static code checks like flake8 and mypy
 	PYEXEFILES="$(git grep -lI '^#!.*python') $(git ls-files "*.py")"
 	flake8 $PYEXEFILES
+
+	mypy --no-error-summary --ignore-missing-imports --scripts-are-modules .
```

Comment:
Does this catch executables automatically? Most of these tools just look for modules. Otherwise this needs to be called on `$PYEXEFILES` as well. Or did you intentionally leave these out yet, as they need some further annotations/fixes?

Reply:
So `--scripts-are-modules` converts "executables" into modules so multiple programs don't conflict (multiple main()'s), but it seems this does not work at all. This is kinda frustrating.... I break image-prune by:

```
diff --git a/image-prune b/image-prune
index d4bde4f..87405e1 100755
--- a/image-prune
+++ b/image-prune
@@ -119,6 +119,9 @@ class ImageCache:

     def prune(self, keepers, force=False, dryrun=False):
         expiry_threshold = time.time() - IMAGE_EXPIRE * 86400
+        foo: bool = 1
+        foo = "str"
+        print(foo)
```

And mypy does not complain at all....

Reply:
But changing to `PYEXEFILES` turns this into a another mess :|

```
[jelle@t14s][~/projects/cockpit-bots]%./test/run
pyproject.toml: Invalid value (at line 10, column 8)
test-failure-policy: error: Duplicate module named "test-failure-policy" (also at "task/test-failure-policy")
test-failure-policy: note: See https://mypy.readthedocs.io/en/stable/running_mypy.html#mapping-file-paths-to-modules for more info
test-failure-policy: note: Common resolutions include: a) using `--exclude` to avoid checking one of them, b) adding `__init__.py` somewhere, c) using `--explicit-package-bases` or adjusting MYPYPATH
```

Reply:
Ok, I just had to sort and unique `$PYEXEFILES` and now it works \o/